internal/item/item_test.go

```diff
@@ -0,0 +1,183 @@
+package item_test
+
+import (
+	"testing"
+
+	"github.com/coralproject/xenia/internal/item"
+	"github.com/coralproject/xenia/internal/item/ifix"
+
+	"github.com/ardanlabs/kit/cfg"
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/db/mongo"
+	"github.com/ardanlabs/kit/tests"
+)
+
+// prefix is what we are looking to delete after the test.
+const prefix = "ITEM_TEST_O"
+
+func init() {
+	// Initialize the configuration and logging systems. Plus anything
+	// else the web app layer needs.
+	tests.Init("CORAL")
+
+	// Initialize MongoDB using the `tests.TestSession` as the name of the
+	// master session.
+	cfg := mongo.Config{
+		Host:     cfg.MustString("MONGO_HOST"),
+		AuthDB:   cfg.MustString("MONGO_AUTHDB"),
+		DB:       cfg.MustString("MONGO_DB") + "_test",
+		User:     cfg.MustString("MONGO_USER"),
+		Password: cfg.MustString("MONGO_PASS"),
+	}
+	tests.InitMongo(cfg)
+}
+
+//==============================================================================
+
+func TestEnsureTypeIndexes(t *testing.T) {
+
+	// get db connection
+	//  should this be moved to a shared testing package?
+	db, err := db.NewMGO(tests.Context, tests.TestSession)
+	if err != nil {
+		t.Fatalf("\t%s\tShould be able to get a Mongo session : %v", tests.Failed, err)
+	}
+	defer db.CloseMGO(tests.Context)
+
+	// register the item types
+	err = ifix.RegisterTypes("types.json")
+	if err != nil {
+		t.Fatalf("\t%s\tCould not register the types from types.json. %s", err)
+	}
+
+	// create the indexes
+	err = item.EnsureTypeIndexes(tests.Context, db, item.Types)
+	if err != nil {
+		t.Fatalf("\t%s\tUnable to create indices", tests.Failed, err)
+
+	}
+
+	//Todo, how can we test to ensure inexes are created?
+
+}
+
+//==============================================================================
+
+func TestRels(t *testing.T) {
+	tests.ResetLog()
+	defer tests.DisplayLog()
+
+	// get db connection
+	//  should this be moved to a shared testing package?
+	db, err := db.NewMGO(tests.Context, tests.TestSession)
+	if err != nil {
+		t.Fatalf("\t%s\tShould be able to get a Mongo session : %v", tests.Failed, err)
+	}
+	defer db.CloseMGO(tests.Context)
+
+	// register the item types
+	err = ifix.RegisterTypes("types.json")
+	if err != nil {
```

Comment:
I'm aware of that, but find the separate lines much more readable. In general, I like "taller and narrower" code.  Easier for me to pattern recognize statements. :)